### ISSUE TITLE
uninit value in Web/Plugin/Device/Ports.pm

### ISSUE THREAD

**Issue by inphobia** [netdisco/netdisco#679]
and that's regretfully all the info i have, 4 entries in netdisco-web.log:
```
Use of uninitialized value in string eq at /home/testdisc/perl5/lib/perl5/App/Netdisco/Web/Plugin/Device/Ports.pm line 197.
Use of uninitialized value in string eq at /home/testdisc/perl5/lib/perl5/App/Netdisco/Web/Plugin/Device/Ports.pm line 197.
Use of uninitialized value in string eq at /home/testdisc/perl5/lib/perl5/App/Netdisco/Web/Plugin/Device/Ports.pm line 197.
Use of uninitialized value in string eq at /home/testdisc/perl5/lib/perl5/App/Netdisco/Web/Plugin/Device/Ports.pm line 197.
```

i tried everything i could think of but cannot find a way to reproduce the error. this most likely happened while working on layer7::netscaler, either lldp or but most likely vlan functions since that was my focus.  yet somehow, even after reverting every commit i still can't find a way to reproduce this. 

i didn't notice any anomalies in the webinterface, so this error is most likely harmless.

## Your Environment

* Netdisco version used: 2.044004
* SNMP::Info version used:  3.70 and then some (https://github.com/inphobia/snmp-info/tree/nn_citrix2)

**Comment by ollyg** [comment]
I don't know how this is happening! There are limits to my Perl knowledge :-).

`$f` is defined otherwise the earlier `eq` would fail with the same error. The `defined or` should check that `$_` is defined before the `eq`. So how is this happening?

**Comment by ollyg** [comment]
https://github.com/netdisco/netdisco/blob/b211c7c7e20c812319f407f348eeb044297c54eb/lib/App/Netdisco/Web/Plugin/Device/Ports.pm#L197

**Comment by ollyg** [comment]
closing as cant reproduce at all, sorry

**Comment by ollyg** [comment]
and as if by magic I tripped over the same error! can't believe my eyes.

**Comment by ollyg** [comment]
fixed in [master b04ffffc] fix #679 unint error
